cmd/kube-scheduler/app/options/insecure_serving.go

```diff
@@ -0,0 +1,240 @@
+
+	controlleroptions "k8s.io/kubernetes/cmd/controller-manager/app/options"
+	schedulerconfig "k8s.io/kubernetes/cmd/kube-scheduler/app/config"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+// CombinedInsecureServingOptions sets up up to two insecure listener for healthz and metrics. The flags
+// override the ComponentConfig and InsecureServingOptions values for both.
+type CombinedInsecureServingOptions struct {
+	Healthz *controlleroptions.InsecureServingOptions
+	Metrics *controlleroptions.InsecureServingOptions
+
+	BindPort    int    // -1 means to ignore, everything else overrides both of the structs above
+	BindAddress string // empty means to ignore, everything else override both of the struct above
+}
+
+// AddFlags adds flags related to insecure serving for controller manager to the specified FlagSet.
+func (s *CombinedInsecureServingOptions) AddFlags(fs *pflag.FlagSet) {
+	if s == nil {
+		return
+	}
+
+	fs.StringVar(&s.BindAddress, "address", s.BindAddress, "DEPRECATED: the IP address on which to listen for the --port port (set to 0.0.0.0 for all IPv4 interfaces and :: for all IPv6 interfaces). See --bind-address instead.")
+	// MarkDeprecated hides the flag from the help. We don't want that:
+	// fs.MarkDeprecated("address", "see --bind-address instead.")
+	fs.IntVar(&s.BindPort, "port", s.BindPort, "DEPRECATED: the port on which to serve HTTP insecurely without authentication and authorization. If 0, don't serve HTTPS at all. See --secure-port instead.")
+	// MarkDeprecated hides the flag from the help. We don't want that:
+	// fs.MarkDeprecated("port", "see --secure-port instead.")
+}
+
+func (o *CombinedInsecureServingOptions) ApplyTo(c *schedulerconfig.Config, componentConfig *componentconfig.KubeSchedulerConfiguration) error {
+	if o == nil {
+		componentConfig.HealthzBindAddress = ""
+		componentConfig.MetricsBindAddress = ""
+		return nil
+	}
+
+	// override ComponentConfig.{Healthz,Metrics}BindAddress with InsecureServing flag value if given.
+	if err := o.applyDeprecatedHealthzAddressToConfig(componentConfig); err != nil {
+		return err
+	}
+	if err := o.applyDeprecatedHealthzPortToConfig(componentConfig); err != nil {
+		return err
+	}
+	if err := syncBindAddressAndInsecureServingOptions(&componentConfig.HealthzBindAddress, o.Healthz); err != nil {
+		return fmt.Errorf("invalid healthz address: %v", err)
+	}
+	if err := syncBindAddressAndInsecureServingOptions(&componentConfig.MetricsBindAddress, o.Metrics); err != nil {
+		return fmt.Errorf("invalid metrics address: %v", err)
+	}
+
+	if err := o.Healthz.ApplyTo(&c.InsecureServing); err != nil {
+		return err
+	}
+	if o.Metrics != nil && (c.ComponentConfig.MetricsBindAddress != c.ComponentConfig.HealthzBindAddress || o.Healthz == nil) {
+		if err := o.Metrics.ApplyTo(&c.InsecureMetricsServing); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func syncBindAddressAndInsecureServingOptions(addr *string, is *controlleroptions.InsecureServingOptions) error {
```

Comment:
Am not sure whether this really useful.

Reply:
Doe not `applyDeprecatedHealthzAddressToConfig` and `applyDeprecatedHealthzPortToConfig` already set the addr and port ?

Reply:
Because we have bad situation that we have a non-modular, non-reusable scheduler component config, and we have componentized option structs. If we split the former in a similar way we can add native ComponentConfigInsecureServing support to the options. Until then we need this sync logic because the flags, Options and component config are sources for the configuration, with certain rules. This leads to quite some legacy complexity.

Reply:
Documented this function and added tests.